gcore/src/msg.rs

```diff
@@ -38,46 +41,76 @@ mod sys {
             data_len: u32,
             value_ptr: *const u8,
             message_id_ptr: *mut u8,
-        );
-        pub fn gr_reply_commit(message_id_ptr: *mut u8, value_ptr: *const u8);
-        pub fn gr_reply_push(data_ptr: *const u8, data_len: u32);
+        ) -> ErrorCode;
+        pub fn gr_reply_commit(message_id_ptr: *mut u8, value_ptr: *const u8) -> ErrorCode;
+        pub fn gr_reply_push(data_ptr: *const u8, data_len: u32) -> ErrorCode;
         pub fn gr_reply_to(dest: *mut u8);
         pub fn gr_send(
             program: *const u8,
             data_ptr: *const u8,
             data_len: u32,
             value_ptr: *const u8,
             message_id_ptr: *mut u8,
-        );
+        ) -> ErrorCode;
         pub fn gr_send_wgas(
             program: *const u8,
             data_ptr: *const u8,
             data_len: u32,
             gas_limit: u64,
             value_ptr: *const u8,
             message_id_ptr: *mut u8,
-        );
+        ) -> ErrorCode;
         pub fn gr_send_commit(
             handle: u32,
             message_id_ptr: *mut u8,
             program: *const u8,
             value_ptr: *const u8,
-        );
+        ) -> ErrorCode;
         pub fn gr_send_commit_wgas(
             handle: u32,
             message_id_ptr: *mut u8,
             program: *const u8,
             gas_limit: u64,
             value_ptr: *const u8,
-        );
-        pub fn gr_send_init() -> u32;
-        pub fn gr_send_push(handle: u32, data_ptr: *const u8, data_len: u32);
+        ) -> ErrorCode;
+        pub fn gr_send_init(handle: *mut u32) -> ErrorCode;
```

Comment:
```suggestion
        pub fn gr_send_init(handle: *mut u8) -> ErrorCode;
```

Can't we go here like that?

Reply:
`u32` is handler's type so it is semantically correct to keep it like that at least